packages/rum/src/domain/record/record.ts

```diff
@@ -119,15 +166,23 @@ export function record(options: RecordOptions): RecordAPI {
         timestamp: timeStampNow(),
       })
     },
+    shadowDomCallBacks: { shadowDomCreatedCallback, shadowDomRemovedCallback },
   })
 
   return {
-    stop: stopObservers,
+    stop: () => {
+      shadowDomCallBacks.forEach(({ stop }) => stop())
+      stopObservers()
+    },
     takeSubsequentFullSnapshot: (timestamp) =>
       takeFullSnapshot(timestamp, {
         status: SerializationContextStatus.SUBSEQUENT_FULL_SNAPSHOT,
         elementsScrollPositions,
       }),
-    flushMutations: () => mutationController.flush(),
+    flushMutations: () => {
+      shadowDomCallBacks.forEach(({ flush }) => flush())
+      documentMutationController.flush()
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏move those lines in a `flushMutations` function to factorize them